api/pkg/s3/objectput.go

```diff
@@ -146,7 +149,13 @@ func (s *APIService) ObjectPut(request *restful.Request, response *restful.Respo
 			log.Debugln("finished read")
 			eof = true
 		}
-		err = stream.Send(&s3.PutObjectRequest{Data: buf[:n]})
+		// encrypt if needed
```

Comment:
Is it better to do encryption in s3 service than in api service?

Reply:
better ? in what way ?

Reply:
IMHO, API service only does parameter checking and request delivery, and s3 service will do encryption when received the request from api service.

Reply:
raising issue #772 to track this